### PR TITLE
DLP V2 Release

### PR DESCRIPTION
**Pull request by jbolinger**
Moving from privately staged version.

### CODE REVIEW

File: google-cloud-dlp/Rakefile [hunk 1]
```diff
@@ -39,65 +39,16 @@ namespace :test do
...
-  # Required for smoke tests
-  ENV["DLP_TEST_PROJECT"] = project
 
   Rake::Task["acceptance:run"].invoke
 end
 
 namespace :acceptance do
   task :run do
-    Rake::Task["smoke_test"].invoke
+    puts "The google-cloud-dlp gem does not have acceptance tests."
```
Comment (blowmage) [inline google-cloud-dlp/Rakefile]: Is the intention to not run these tests as part of the CI build?

File: google-cloud-dlp/Rakefile [hunk 2]
```diff
@@ -124,11 +75,11 @@ task :jsondoc => :yard do
     documents: [
       {
         type: "toc",
-        title: "Google::Privacy::Dlp::V2beta1::DataTypes",
+        title: "Google::Privacy::Dlp::V2::DataTypes",
```
Comment (blowmage) [inline google-cloud-dlp/Rakefile]: Do you mean to replace `V2beta1` with `V2`, or do you want to show both `V2` and `V2beta1` versions in the documentation? It looks like the `V2beta1` files are not being removed, right?

File: google-cloud-dlp/lib/google/cloud/dlp.rb [hunk 3]
```diff
@@ -37,21 +35,6 @@ module Cloud
     # 3. [Enable the DLP API.](https://console.cloud.google.com/apis/api/dlp)
     # 4. [Setup Authentication.](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/master/guides/authentication)
     #
-    # ### Preview
```
Comment (blowmage) [inline google-cloud-dlp/lib/google/cloud/dlp.rb]: Do you want this code example removed?

File: google-cloud-dlp/Gemfile [hunk 4]
```diff
@@ -6,13 +6,6 @@ gem "gcloud-jsondoc",
     git: "https://github.com/GoogleCloudPlatform/google-cloud-ruby.git",
     branch: "gcloud-jsondoc"
 
-if Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.1")
```
Comment (blowmage) [inline google-cloud-dlp/Gemfile]: I think we still need this conditional in the Gemfile in order to support Ruby 2.0 in the builds.

File: google-cloud-dlp/README.md [hunk 5]
```diff
@@ -21,32 +19,6 @@ steps:
 $ gem install google-cloud-dlp
 ```
 
-### Supported Ruby Versions
```
Comment (blowmage) [inline google-cloud-dlp/README.md]: I believe we want to keep this info in the README.

**Comment by jbolinger** [comment]
Thanks @blowmage and sorry for the noise. I'm still wrapping my head around how to keep in the manual changes when we regenerate.

I'm not sure about the doc links - @geigerj do you know?

File: google-cloud-dlp/.rubocop.yml [hunk 6]
```diff
@@ -4,7 +4,7 @@ AllCops:
     - "lib/google/**/*"
     - "Rakefile"
     - "test/**/*"
-    - "acceptance/google/cloud/dlp/v2beta1/dlp_service_smoke_test.rb"
+    - "acceptance/google/cloud/dlp/v2/dlp_service_smoke_test.rb"
```
Comment (blowmage) [inline google-cloud-dlp/.rubocop.yml]: If you want to run both tests it might be a good idea to make this `"acceptance/**/*smoke_test.rb"`

**Comment by geigerj** [comment]
@jbolinger The doc links need to be maintained manually for gems with multiple supported server versions. (Generally you want to retain the links to the previous version until it is no longer supported.)

It is also surprising that there is an acceptance tests for the previous version but not the new version. That might point to an upstream configuration issue (missing smoke test config?).

**Comment by quartzmo** [comment]
@jbolinger How's it going, do you need help with any manual changes?

File: google-cloud-dlp/docs/toc.json [hunk 7]
```diff
@@ -1,6 +1,28 @@
 {
   "guides": [],
   "services": [
+    {
```
Comment (blowmage) [inline google-cloud-dlp/docs/toc.json]: I believe we want to define one service, but have both `V2` and `V2beta1` listed under `nav`.

Comment (blowmage) [reply google-cloud-dlp/docs/toc.json]: @quartzmo ^ Can you confirm?

**Comment by jbolinger** [comment]
Thanks for checking up on this @quartzmo! I've done all I can think to do on this one, but I don't know what I don't know. @blowmage pointed out one thing we need to clarify before we can finish. Beyond that, I don't know of anything else that needs to be done.

File: google-cloud-dlp/docs/toc.json (hunk 7 above)
Comment (quartzmo) [reply google-cloud-dlp/docs/toc.json]: Sorry, just saw this, taking a look now.

**Comment by quartzmo** [comment]
I get these errors when running `rake acceptance` locally:

```sh

  1) Error:
DlpServiceSmokeTest#test_0001_runs one smoke test with inspect_content:
ArgumentError: wrong number of arguments (given 2, expected 1)
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-dlp/lib/google/cloud/dlp/v2/dlp_service_client.rb:445:in `inspect_content'
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-dlp/acceptance/google/cloud/dlp/v2/dlp_service_smoke_test.rb:32:in `block (2 levels) in <top (required)>'

  2) Error:
DlpServiceSmokeTest#test_0001_runs one smoke test with inspect_content:
ArgumentError: wrong number of arguments (given 2, expected 1)
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-dlp/lib/google/cloud/dlp/v2/dlp_service_client.rb:445:in `inspect_content'
    /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-dlp/acceptance/google/cloud/dlp/v2beta1/dlp_service_smoke_test.rb:32:in `block (2 levels) in <top (required)>'

2 runs, 0 assertions, 0 failures, 2 errors, 0 skips
```

File: google-cloud-dlp/docs/toc.json (hunk 7 above)
Comment (quartzmo) [reply google-cloud-dlp/docs/toc.json]: @jbolinger Yes, @blowmage is correct. Can you change this to the structure used in [google-cloud-video_intelligence/docs/toc.json](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-video_intelligence/docs/toc.json)?

Comment (jbolinger) [reply google-cloud-dlp/docs/toc.json]: No problem, moved to nav.

**Comment by jbolinger** [comment]
@quartzmo I fixed the test manually. Thanks for pointing that out, but I'm curious why don't they run in CI? 

There is no actual test for the v2 service and it looks like the API changed. I'm going to check with the team about that. I assume we don't write those ourselves - correct?

**Comment by quartzmo** [comment]
The acceptance tests don’t run in CI for PRs for security reasons, isn’t that correct @blowmage?

File: google-cloud-dlp/acceptance/google/cloud/dlp/v2beta1/dlp_service_smoke_test.rb [hunk 8]
```diff
@@ -17,12 +17,12 @@
...
 require "minitest/spec"
 
-require "google/cloud/dlp"
+require "google/cloud/dlp/v2beta1"
 
 describe "DlpServiceSmokeTest" do
   it "runs one smoke test with inspect_content" do
 
-    dlp_service_client = Google::Cloud::Dlp.new
+    dlp_service_client = Google::Cloud::Dlp::V2beta1::DlpServiceClient.new
```
Comment (blowmage) [inline google-cloud-dlp/acceptance/google/cloud/dlp/v2beta1/dlp_service_smoke_test.rb]: Would it be a good idea to keep the original require and call `Google::Cloud::Dlp.new version: :v2beta1` here?

**Comment by blowmage** [comment]
That is correct. Acceptance tests only run on master.

File: google-cloud-dlp/acceptance/google/cloud/dlp/v2beta1/dlp_service_smoke_test.rb (hunk 8 above)
Comment (jbolinger) [reply google-cloud-dlp/acceptance/google/cloud/dlp/v2beta1/dlp_service_smoke_test.rb]: Great suggestion - done.

**Comment by quartzmo** [comment]
> There is no actual test for the v2 service... I assume we don't write those ourselves - correct?

I'm not certain about that. I usually look to Video Intelligence as an example of how a GAPIC-only package should be done. @geigerj Did you write [video_intelligence_service_smoke_test.rb](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-video_intelligence/acceptance/google/cloud/video_intelligence/v1/video_intelligence_service_smoke_test.rb) by hand?

**Comment by geigerj** [comment]
No, the smoke tests are autogenerated. @blowmage is probably correct that the generator needs a fix to support smoke tests for side-by-side server versions.

The configuration for the v2beta1 smoke test is here: https://github.com/googleapis/googleapis/blob/master/google/privacy/dlp/v2beta1/dlp_gapic.yaml#L25. It looks like the config is missing for v2: https://github.com/googleapis/googleapis/blob/master/google/privacy/dlp/v2/dlp_gapic.yaml.

**Comment by quartzmo** [comment]
The smoke tests pass for me locally.

File: google-cloud-dlp/Rakefile [hunk 9]
```diff
@@ -139,6 +139,24 @@ task :jsondoc => :yard do
...
           }
         ]
+      },
+      {
+        type: "toc",
+        title: "Google::Privacy::Dlp::V2::DataTypes",
+        modules: [
+          {
+            title: "Google::Privacy::Dlp::V2",
+            include: ["google/cloud/dlp/v2"]
```
Comment (quartzmo) [inline google-cloud-dlp/Rakefile]: The `include` pattern for the GAPIC code in "Data Types" needs to end in a slash `/`, or the output will include content from the `V2` module and end up broken with a table inside a table. The [Video Intelligence Rakefile](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-video_intelligence/Rakefile#L132) does it correctly. Is this something that can be fixed in the generation? We just had this problem with BigQuery Data Transfer as well.

Line 131, although not part of this PR, needs the same fix, for `V2beta1`.

**Comment by blowmage** [comment]
Both acceptance tests also pass for me locally.

File: google-cloud-dlp/Rakefile (hunk 9 above)
Comment (quartzmo) [reply google-cloud-dlp/Rakefile]: Actually, @blowmage pointed out that other than Video Intelligence and BQ Data Transfer, most GAPIC packages do *not* have the trailing slash, and just show the module docs (the `V2` docs, in this case) at the Data Types page. Let's hold off on changing this for now and see what happens. If I get the chance, I'll try to run the docs locally to check them first, but it shouldn't hold up the PR.

Comment (jbolinger) [reply google-cloud-dlp/Rakefile]: Whoops, just noticed your second comment. I'll revert it.

Comment (quartzmo) [reply google-cloud-dlp/Rakefile]: Sorry about that!

**Comment by jbolinger** [comment]
Thanks for all your help @quartzmo and @blowmage! They would like the release to go out before the end of the week so I'll take a crack at it tomorrow morning. Bear with me and let me know if you'd like anything done a certain way. The guide looks straightforward, but it'll be the first time 😄

**Comment by blowmage** [comment]
You will probably want to verify the docs generate properly before cutting a release.

**Pull request merged**